.drone.yml

```diff
@@ -1,24 +1,55 @@
 ---
 kind: pipeline
+type: docker
 name: testing
 
 platform:
   os: linux
   arch: amd64
 
 steps:
+- name: environment
+  pull: always
+  image: golang:1.15
+  commands:
+  - go version
+  - go env
+  volumes:
+  - name: gopath
+    path: /go
+
+- name: staticcheck
+  pull: never
+  image: golang:1.15
+  commands:
+  - go run honnef.co/go/tools/cmd/staticcheck ./...
+  volumes:
+  - name: gopath
+    path: /go
+
+- name: lint
+  pull: never
+  image: golang:1.15
+  commands:
+  - go run golang.org/x/lint/golint -set_exit_status ./...
+  volumes:
+  - name: gopath
+    path: /go
+
 - name: vet
-  image: golang:1.12
+  pull: never
+  image: golang:1.15
   commands:
   - go vet ./...
   volumes:
   - name: gopath
     path: /go
 
 - name: test
-  image: golang:1.12
+  pull: never
```

Comment:
Never doesn't sound like a good idea

Reply:
The first one is an `always` which will get any changes. The rest are `never` because the environment was set in the first one.

Reply:
we should always favor `pull: if-not-exists` over `pull: never` because they are basically the same behavior, and it prevents a possible race condition.  However, in this case we can remove the `pull` attribute all together because tagged images use `if-not-exists` by default.